Source/WebKit/webpushd/WebPushDaemon.mm

```diff
@@ -416,21 +416,52 @@ static void tryCloseRequestConnection(xpc_object_t request)
         RELEASE_LOG_ERROR_IF(cfError, Push, "Failed to launch process in response to push: %{public}@", (__bridge NSError *)cfError);
     });
 #elif PLATFORM(IOS) || PLATFORM(VISION)
-    const NSString *URLPrefix = @"webapp://web-push/";
-    NSURL *launchURL = [NSURL URLWithString:[URLPrefix stringByAppendingFormat:@"%@", (NSString *)subscriptionSetIdentifier.pushPartition]];
+    if (bundleIdentifier == "com.apple.SafariViewService"_s) {
+        const NSString *URLPrefix = @"webapp://web-push/";
+        NSURL *launchURL = [NSURL URLWithString:[URLPrefix stringByAppendingFormat:@"%@", (NSString *)subscriptionSetIdentifier.pushPartition]];
+
+        NSDictionary *options = @{
+            FBSOpenApplicationOptionKeyActivateForEvent: @{ FBSActivateForEventOptionTypeBackgroundContentFetching: @{ } },
+            FBSOpenApplicationOptionKeyPayloadURL : launchURL,
+            FBSOpenApplicationOptionKeyPayloadOptions : @{ UIApplicationLaunchOptionsSourceApplicationKey : @"com.apple.WebKit.webpushd" },
+        };
+
+        RetainPtr configuration = adoptNS([[_LSOpenConfiguration alloc] init]);
+        configuration.get().sensitive = YES;
+        configuration.get().frontBoardOptions = options;
+        configuration.get().allowURLOverrides = NO;
+
+        [[LSApplicationWorkspace defaultWorkspace] openURL:launchURL configuration:configuration.get() completionHandler:^(NSDictionary<NSString *, id> *result, NSError *error) {
+            if (error)
+                RELEASE_LOG_ERROR(Push, "Failed to open app to handle push: %{public}@", error);
+        }];
 
-    NSDictionary *options = @{
-        FBSOpenApplicationOptionKeyActivateForEvent: @{ FBSActivateForEventOptionTypeBackgroundContentFetching: @{ } },
-        FBSOpenApplicationOptionKeyPayloadURL : launchURL,
-        FBSOpenApplicationOptionKeyPayloadOptions : @{ UIApplicationLaunchOptionsSourceApplicationKey : @"com.apple.WebKit.webpushd" },
+        return;
+    }
+
+    NSDictionary *settingsInfo = @{
+        pushActionVersionKey(): currentPushActionVersion(),
+        pushActionPartitionKey(): (NSString *)subscriptionSetIdentifier.pushPartition,
```

Comment:
maybe add an early bailout and `RELEASE_LOG_ERROR` is `pushPartition` is somehow nil